src/main/java/ui/MenuControl.java

```diff
@@ -89,47 +87,71 @@ private Menu createFileMenu() {
     }
 
     /**
-     * Logs the user action and creates a new board
+     * Checks whether the current open board is dirty, i.e. has unsaved changes.
+     *
+     * @return true if current open board is dirty, false otherwise
      */
-    private void onBoardNew() {
-        logger.info("Menu: Boards > New");
-        newBoard();
+    public final boolean isCurrentOpenBoardDirty() {
```

Comment:
I think `isCurrentBoardDirty` is sufficient as we only have 1 open board. Update header comment as well.